### ISSUE TITLE
Machine Learning support for constructing maps from scratch in Newt

### ISSUE THREAD

**Issue by ugurdogrusoz** [nrnb/GoogleSummerOfCode#231]
### Background
[Newt](https://newteditor.org/) [1] is a [Cytoscape.js](http://js.cytoscape.org/) [2] based, open source viewer and editor for pathways in [Systems Biological Graphical Notation (SBGN)](http://sbgn.org/) [3] and [Simple Interaction Format (SIF)](http://www.pathwaycommons.org/pc2/formats#sif_relations). It's been built (with partial support from GSoC) around an editing library named [ChiSE](https://github.com/iVis-at-Bilkent/chise.js), which in turn was based on a viewer library named [SBGNViz](https://github.com/iVis-at-Bilkent/sbgnviz.js) [4]. The tool uses a number of [Cytoscape.js extensions](http://js.cytoscape.org/#extensions) built for various editing capabilities such as grid-guidelines, context-menus, and expand-collapse.

[1] H. Balci et al., "[Newt: a comprehensive web-based tool for viewing, constructing, and analysing biological maps](https://doi.org/10.1093/bioinformatics/btaa850)", Bioinformatics, 37(10), pp.1475-1477, 2021.
[2] M. Franz et al. "[Cytoscape. js 2023 update: a graph theory library for visualization and analysis.](https://academic.oup.com/bioinformatics/article/39/1/btad031/6988031)", Bioinformatics 39.1, btad031, 2023.
[3] N. Le Novère et al., "[The Systems Biology Graphical Notation](http://www.nature.com/nbt/journal/v27/n8/abs/nbt.1558.html)", Nature Biotechnology, 27, pp. 735-741, 2009.
[4] M. Sari et al., "[SBGNViz: a tool for visualisation and complexity management of SBGN process description maps](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0128985)", PLoS ONE, 10(6), e0128985, 2015.

### Goal
The goal of this project is to enable users to describe the interactions in a pathway textually rather than having to learn how to use Newt and its strict notation. Then an ML-based agent will interpret this text and produce the corresponding partial map. The user may or may not add the suggested partial map to the current drawing.

### Difficulty Level: Hard
This project will require learning several repositories as well as some new ML components built for constructing pathways.

### Size and Length of Project
- **large: 350 hours**

### Skills
Essential skills: JavaScript, HTML, CSS, Java
Nice to have skills: Cytoscape.js and ML libraries

### Public Repository
[Newt GitHub repository](https://github.com/iVis-at-Bilkent/newt)
[ChiSE GitHub repository](https://github.com/iVis-at-Bilkent/chise.js)
[SBGNViz.js GitHub repository](https://github.com/iVis-at-Bilkent/sbgnviz.js)

### Potential Mentors
Ugur Dogrusoz
Adrien Rougny

**Comment by vigasinii** [comment]
Greetings Ugur, @ugurdogrusoz,
I'm excited about the opportunity to contribute to this issue and believe I can offer valuable insights to assist in its resolution. Could you kindly guide me to the suitable Slack server for the organization? I'd appreciate the opportunity to discuss any questions I may have about this issue there.

**Comment by khanspers** [comment]
NRNB has been accepted as a mentoring organization for GSoC 2024. The contributor application period if March 28 – April 2. Here are some useful links:

[GSoC contributor guide](https://google.github.io/gsocguides/student/)
[NRNB project proposal template](https://docs.google.com/document/d/1Zi6L38CHEeq2aL6xzv0Ozhd_Y6D71W3yCBGHplmxr6k/edit?usp=sharing)
[Eligibility requirements](https://developers.google.com/open-source/gsoc/faq#what_are_the_eligibility_requirements_for_participation)
[Full program timeline](https://developers.google.com/open-source/gsoc/timeline)

**Comment by 143divya** [comment]
is this project still open?. because  i want to work on this project

**Comment by Ankita3503** [comment]
Hi @ugurdogrusoz . I am excited to contribute to your project . Can you please guide me how to leverage my knowledge in Cytoscape.js extensions . It will be very helpful in contributing the project  before starting my work in this new tech.

Hoping to hear from you .

**Comment by renu-123-pixel** [comment]
i want to work on this issue . please assign me this issue.

**Comment by khanspers** [comment]
This is an active GSoC 2024 project. Closing this project idea as it is no longer available to other contributors.
